service/controller/resource/monitoring/remotewriteapiendpointconfigsecret/create.go

```diff
@@ -36,20 +36,6 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		}
 
 		if current != nil {
-			// We thought that having an immutable secret would be a good thing as the remote write password cannot be changed (causing remote write errors)
-			// However, this causes a lot of issues if we want to update the other configurations like the queue config.
-			// Hence if the secret is immutable, we force delete it to create a non-immutable one
-			if current.Immutable != nil && *current.Immutable {
-				err = r.deleteSecret(ctx, current)
-				if err != nil {
-					return microerror.Mask(err)
-				}
-				err = r.createSecret(ctx, cluster, name, namespace)
-				if err != nil {
-					return microerror.Mask(err)
-				}
-			}
-
```

Comment:
On the first deployment, who create the secret ?

Reply:
We initially created the secret as immutable, this code killed it and creates a non immutable secret. It was deployed everywhere so we do not use immutable secrets anymore

Reply:
Ok, but I'm wondering what will happen on a new installation ? the secret will not be created at all, right?

Reply:
https://github.com/giantswarm/prometheus-meta-operator/blob/a8c64346aa4ba0fd04539f491a5b17cfe4cdda8f/service/controller/resource/monitoring/remotewriteapiendpointconfigsecret/create.go#L30

Reply:
![image](https://user-images.githubusercontent.com/2787548/228226287-94dd6ead-82c0-4e9c-b213-1213ad592b57.png)